src/abb_robot_client_RWS2/rws.py

```diff
@@ -0,0 +1,1124 @@
+        Reset RAPID program pointer to main in normal tasks
+        """
+        res=self._do_post("rw/rapid/execution/resetpp") 
+
+    def get_ramdisk_path(self) -> str: #TODO: I have know idea what this is for or how to replicate it
+        """
+        Get the path of the RAMDISK variable on the controller
+
+        :return: The RAMDISK path
+        """
+        res_json = self._do_get("ctrl/$RAMDISK")
+        return res_json["state"][0]["value"]
+
+    def get_execution_state(self) -> RAPIDExecutionState: #
+        """
+        Get the RAPID execution state
+
+        :return: The RAPID execution state
+        """
+        res_json = self._do_get("rw/rapid/execution")
+        print(res_json)
+        state = res_json["state"][0]
+        ctrlexecstate=state["ctrlexecstate"]
+        cycle=state["cycle"]
+        return RAPIDExecutionState(ctrlexecstate, cycle)
+    
+    def get_controller_state(self) -> str:#
+        """
+        Get the controller state. The controller state can have the following values:
+
+        `init`, `motoroff`, `motoron`, `guardstop`, `emergencystop`, `emergencystopreset`, `sysfail`
+
+        RAPID can only be executed and the robot can only be moved in the `motoron` state.
+
+        :return: The controller state
+        """
+        res_json = self._do_get("rw/panel/ctrl-state")
+        state = res_json["state"][0]
+        return state['ctrlstate']
+
+    def set_controller_state(self, ctrl_state): #
+        payload = {"ctrl-state": ctrl_state}
+        res=self._do_post("rw/panel/ctrl-state?action=setctrlstate", payload)
+    
+    def get_operation_mode(self) -> str:#
+        """
+        Get the controller operational mode. The controller operational mode can have the following values:
+
+        `INIT`, `AUTO_CH`, `MANF_CH`, `MANR`, `MANF`, `AUTO`, `UNDEF`
+
+        Typical values returned by the controller are `AUTO` for auto mode, and `MANR` for manual reduced-speed mode.
+        
+        :return: The controller operational mode.
+        """
+        res_json = self._do_get("rw/panel/opmode")        
+        state = res_json["state"][0]
+        return state["opmode"]
+    
+    def get_digital_io(self, signal: str, network: str='Local', unit: str='DRV_1') -> int:#
+        """
+        Get the value of a digital IO signal.
+
+        :param signal: The name of the signal
+        :param network: The network the signal is on. The default `Local` will work for most signals.
+        :param unit: The drive unit of the signal. The default `DRV_1` will work for most signals.
+        :return: The value of the signal. Typically 1 for ON and 0 for OFF
+        """
+        res_json = self._do_get("rw/iosystem/signals/" + network + "/" + unit + "/" + signal)
+        state = res_json["_embedded"]["resources"][0]["lvalue"]
+        return int(state)
+    
+    def set_digital_io(self, signal: str, value: Union[bool,int], network: str='Local', unit: str='DRV_1'): #TODO: Change defaults?

```

Comment:
Relative to your TODO comment, why would you want default values in a function like this? Seems weird that the initial implementation has them, but maybe DRV_1 is a common automatically generated digital IO value?